gitlab-runner/examples/demo-runner.rs

```diff
@@ -187,18 +188,30 @@ impl JobHandler<DemoFile> for Run {
 }
 
 #[tokio::main]
-async fn main() {
+async fn main() -> Result<()> {
     let opts = Opts::from_args();
-    let dir = tempfile::tempdir().unwrap();
+    let dir = tempfile::tempdir().context("Failed to create temporary directory for builds")?;
 
     let (layer, jobs) = GitlabLayer::new();
 
+    let env = std::env::var("RUNNER_LOG").unwrap_or_else(|_| "demo_runner=info,warn".to_string());
+
+    // Always enable the gitlab_runner::job span specifically as provides the gitlab.job field to
+    // correlate gitlab jobs to log entries
+    let envfilter = tracing_subscriber::EnvFilter::builder()
+        .parse(env)
+        .context("Failed to parse env filter: {filter}")?
```

Comment:
Also missing `format!` here?

```suggestion
        .context(format!("Failed to parse env filter: {filter}"))?
```

Reply:
same as the other `.with_context(|| format!(..))`